test/common/PrepDataFuncs.cpp

```diff
@@ -341,6 +342,50 @@ namespace RcclUnitTesting
     return TEST_SUCCESS;
   }
 
+  ErrCode DefaultPrepData_AllToAllv(CollectiveArgs &collArgs)
+  {
+
+    CHECK_CALL(CheckAllocation(collArgs));
+    size_t const numInputBytes = collArgs.numInputElements * DataTypeToBytes(collArgs.dataType);
+    size_t const numOutputBytes = collArgs.numOutputElements * DataTypeToBytes(collArgs.dataType);
+
+    // calculating maxBytes as the maximum number of input bytes out of all the ranks
+    size_t maxBytes = 0;
+    for (int baseRank = 0; baseRank < collArgs.totalRanks; ++baseRank)
+    for (int offsetRank = 0; offsetRank < collArgs.totalRanks; ++offsetRank)
+    {
+      size_t rankSendCount = collArgs.options.sdispls[(baseRank)*collArgs.totalRanks+offsetRank] + collArgs.options.sendcounts[(baseRank)*collArgs.totalRanks+offsetRank];
+      if (maxBytes < rankSendCount)
+        maxBytes = rankSendCount;
+    }
+    maxBytes = maxBytes*DataTypeToBytes(collArgs.dataType);
+
+    // Clear outputs on all ranks (prior to input in case of in-place)
+    collArgs.outputGpu.ClearGpuMem(numOutputBytes);
+
+    // Generate input on root rank - each rank will receive a portion
+    PtrUnion tempInput;
+    tempInput.AllocateCpuMem(maxBytes);
+
+    for (int sendRank = 0; sendRank < collArgs.totalRanks; ++sendRank)
+    {
+      tempInput.FillPattern(collArgs.dataType, maxBytes/DataTypeToBytes(collArgs.dataType), sendRank, false);
+
+      size_t recvDspls = collArgs.options.rdispls[collArgs.globalRank*collArgs.totalRanks + sendRank] * DataTypeToBytes(collArgs.dataType);
+      size_t rankDspls = collArgs.options.sdispls[sendRank*collArgs.totalRanks + collArgs.globalRank] * DataTypeToBytes(collArgs.dataType);
```

Comment:
Prefer sendDspls over rankDspls, although these shoudl have been renamed sendOffsets/recvOffsets.

Reply:
The Dspls is there following the way the allToAllv api is setup in https://github.com/ROCmSoftwarePlatform/rccl/blob/develop/src/collectives/all_to_allv_api.cc
I felt like keeping it more all the same might help future maintenance